pyproject.toml

```
@@ -0,0 +1,299 @@
+# Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+# Copyright (c) 2025 Munich Quantum Software Company GmbH
+# All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+#
+# Licensed under the MIT License
+
+[build-system]
+requires = ["scikit-build-core>=0.11.6", "setuptools-scm>=9.2.2"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "core-plugins-catalyst"
+description = "An MLIR-plugin for connecting MQT Core with Xanadu's Catalyst"
+readme = "README.md"
+authors = [
+    { name = "Lukas Burgholzer", email = "burgholzer@me.com" }
+]
+keywords = ["MQT", "quantum-computing", "design-automation", "Catalyst", "MLIR", "plugin"]
+license = "MIT"
+license-files = ["LICENSE.md"]
+
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Science/Research",
+  "Natural Language :: English",
+  "Operating System :: MacOS",
+  "Operating System :: Microsoft :: Windows",
+  "Operating System :: POSIX :: Linux",
+  "Programming Language :: C++",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Scientific/Engineering :: Electronic Design Automation (EDA)",
+  "Typing :: Typed",
+]
+requires-python = ">=3.11"
+dynamic = ["version"]
+dependencies = [
+  "mqt.core>=3.3.3",
```

```suggestion
  "mqt.core~=3.3.3",
```
Pretty sure we need to upper cap here